### PR TITLE
DOPE-228: added Between and NotBetween comparison and extension functions

### PR DESCRIPTION
**Pull request by martinagallati-ergon**

### CODE REVIEW

**Review by jansigi — APPROVED** [review]

**Review by m0ne — APPROVED** [review]

**Review by jansigi — APPROVED** [review]

File: core/src/test/kotlin/ch/ergon/dope/operators/comparison/BetweenExpressionTest.kt
Comment (pgruntz) [inline core/src/test/kotlin/ch/ergon/dope/operators/comparison/BetweenExpressionTest.kt:1]: please add a test where you actually use it in a query

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/BetweenExpression.kt
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/BetweenExpression.kt:1]: Might be a good idea to merge these Between/NotBetween, IsNull/IsNotNull, Like/NotLike into one file each

File: core/src/test/kotlin/ch/ergon/dope/operators/comparison/BetweenExpressionTest.kt
Comment (martinagallati-ergon) [reply core/src/test/kotlin/ch/ergon/dope/operators/comparison/BetweenExpressionTest.kt:1]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/BetweenExpression.kt
Comment (martinagallati-ergon) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/BetweenExpression.kt:1]: done

**Pull request merged**